workers/generic-worker/errorreport/errorreport.go

```diff
@@ -0,0 +1,28 @@
+package errorreport
+
+import (
+	"github.com/taskcluster/taskcluster/v30/internal/workerproto"
+)
+
+func Send(proto *workerproto.Protocol, err error, debugInfo map[string]string) {
+	title := "generic-worker error"
+	description := err.Error()
+	// could support differentiating for panics
+	kind := "worker-error"
+	// convert debugInfo from map[string]string to map[string]interface{}
+	extra := map[string]interface{}{}
+	for k, v := range debugInfo {
+		extra[k] = v
+	}
+	if proto.Capable("error-report") {
```

Comment:
nit: it would be more efficient to check this condition at the start of the function, e.g. at the start have instead:
```go
	if !proto.Capable("error-report") {
		return
	}
```
and then here have one less level of nesting without the `if` statement.